### DISCUSSION TITLE
Always `follow` flake-parts' `nixpkgs-lib` input?

### DISCUSSION THREAD

**Discussion by srid** [hercules-ci/flake-parts#247] [Q&A]
>[!TIP]
> tldr; Why not do this in _every_ repo that uses flake-parts ? https://github.com/srid/haskell-template/commit/b7c66062535f936afbd7ee4b9440a56ce923903f

flake-parts has its own nixpkgs input:
https://github.com/hercules-ci/flake-parts/blob/567b938d64d4b4112ee253b9274472dc3a346eb6/flake.nix#L5

While it calls it `nixpkgs-lib`, the actual input refers to the whole nixpkgs repo which is pretty large. This leads to [yet another](https://x.com/sridca/status/1830656321349415386) fetch of nixpkgs thus slowing down (due to fetch and extract of nixpkgs) the initial eval of nix commands like `nix develop`.

Would it not make sense for user flakes to do the following always?

```nix
    flake-parts.url = "github:hercules-ci/flake-parts";
    flake-parts.inputs.nixpkgs-lib.follows = "nixpkgs";
```

Thus reducing the number of `nixpkgs`s in the `flake.lock` file, viz.:

```diff
diff --git a/flake.lock b/flake.lock
index 7ebc7b7..8e74a25 100644
--- a/flake.lock
+++ b/flake.lock
@@ -2,7 +2,9 @@
   "nodes": {
     "flake-parts": {
       "inputs": {
-        "nixpkgs-lib": "nixpkgs-lib"
+        "nixpkgs-lib": [
+          "nixpkgs"
+        ]
       },
       "locked": {
         "lastModified": 1722555600,
@@ -64,18 +66,6 @@
         "type": "github"
       }
     },
-    "nixpkgs-lib": {
-      "locked": {
-        "lastModified": 1722555339,
-        "narHash": "sha256-uFf2QeW7eAHlYXuDktm9c25OxOyCoUOQmh5SZ9amE5Q=",
-        "type": "tarball",
-        "url": "https://github.com/NixOS/nixpkgs/archive/a5d394176e64ab29c852d03346c1fc9b0b7d33eb.tar.gz"
-      },
-      "original": {
-        "type": "tarball",
-        "url": "https://github.com/NixOS/nixpkgs/archive/a5d394176e64ab29c852d03346c1fc9b0b7d33eb.tar.gz"
-      }
-    },
     "root": {
       "inputs": {
         "flake-parts": "flake-parts",

```

**Comment by brsvh** [comment]
Maybe user should be prompted to do so in the **Getting Started** section of the document and in the default template.

https://github.com/hercules-ci/flake-parts/blob/3d04084d54bedc3d6b8b736c70ef449225c361b1/template/default/flake.nix#L5-L6

**Comment by roberth** [comment]
It wasn't always, but now it's only the lib tree of Nixpkgs.

```console
$ curl -L https://github.com/NixOS/nixpkgs/archive/a5d394176e64ab29c852d03346c1fc9b0b7d33eb.tar.gz | wc -c
[...]
244736
```

=> quarter of a megabyte.
